scripts/remote-dev-context7-device-login.py

```
@@ -0,0 +1,468 @@
+    access_token = decoded.get("access_token")
+    token_type = decoded.get("token_type")
+    if (
+        not isinstance(access_token, str)
+        or not access_token.startswith(CONTEXT7_KEY_PREFIX)
+        or len(access_token) <= len(CONTEXT7_KEY_PREFIX)
+    ):
+        raise DeviceLoginError("Context7 login did not return the expected long-lived API-key format")
+    if not isinstance(token_type, str) or token_type.lower() != "bearer":
+        raise DeviceLoginError("Context7 login returned an unexpected credential type")
+    if decoded.get("refresh_token"):
+        raise DeviceLoginError("Context7 login returned refresh-token state instead of the reviewed API-key flow")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)